pkg/sethandler/controller.go

```diff
@@ -104,7 +104,9 @@ func (setHandler *SetHandler) PodAdded(pod v1.Pod) {
 	if !shouldPodBeHandled(pod) {
 		return
 	}
-	setHandler.adjustContainerSets(pod)
+	if proceed, containersToBeSet := determineContainersToBeSet(v1.Pod{}, pod); proceed {
```

Comment:
is there a reason to call this during Pod Add?
I mean all containers need to be adjusted here for sure, right?

Reply:
I mean you can just blindly create a set from all containers present and pass it to adjustContainerSets
at the same time internals of determineContainersToBeSet could be greatly streamlined if you assume you only need to deal with UPDATE operation (?)